{"ast":null,"code":"import axios from \"axios\";\nimport { toast } from \"react-toastify\";\n\nconst successCallBack = success => {\n  consolle.log(success.data.message);\n  toast(success.data.message);\n};\n\nconst errorCallBack = error => {\n  const expectedError = error.response && error.response.status >= 404 && error.response.status < 500;\n\n  if (!expectedError) {\n    console.log(\"logging the error\", error);\n    toast.error(\"an unexpected error occured\");\n  }\n\n  return Promise.reject(error);\n}; //intercept errors while communicating with backend server\n\n\naxios.interceptors.response.use(successCallBack, errorCallBack);\nexport default {\n  get: axios.get,\n  post: axios.post,\n  put: axios.put,\n  delete: axios.delete\n};","map":{"version":3,"sources":["/Users/prashantkansal/Desktop/techgropse/digiTender/digiTenderFrontend/src/services/httpService.js"],"names":["axios","toast","successCallBack","success","consolle","log","data","message","errorCallBack","error","expectedError","response","status","console","Promise","reject","interceptors","use","get","post","put","delete"],"mappings":"AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,KAAT,QAAsB,gBAAtB;;AACA,MAAMC,eAAe,GAAIC,OAAD,IAAa;AACnCC,EAAAA,QAAQ,CAACC,GAAT,CAAaF,OAAO,CAACG,IAAR,CAAaC,OAA1B;AACAN,EAAAA,KAAK,CAACE,OAAO,CAACG,IAAR,CAAaC,OAAd,CAAL;AACD,CAHD;;AAKA,MAAMC,aAAa,GAAIC,KAAD,IAAW;AAC/B,QAAMC,aAAa,GACjBD,KAAK,CAACE,QAAN,IACAF,KAAK,CAACE,QAAN,CAAeC,MAAf,IAAyB,GADzB,IAEAH,KAAK,CAACE,QAAN,CAAeC,MAAf,GAAwB,GAH1B;;AAIA,MAAI,CAACF,aAAL,EAAoB;AAClBG,IAAAA,OAAO,CAACR,GAAR,CAAY,mBAAZ,EAAiCI,KAAjC;AACAR,IAAAA,KAAK,CAACQ,KAAN,CAAY,6BAAZ;AACD;;AACD,SAAOK,OAAO,CAACC,MAAR,CAAeN,KAAf,CAAP;AACD,CAVD,C,CAYA;;;AACAT,KAAK,CAACgB,YAAN,CAAmBL,QAAnB,CAA4BM,GAA5B,CAAgCf,eAAhC,EAAiDM,aAAjD;AAEA,eAAe;AACbU,EAAAA,GAAG,EAAElB,KAAK,CAACkB,GADE;AAEbC,EAAAA,IAAI,EAAEnB,KAAK,CAACmB,IAFC;AAGbC,EAAAA,GAAG,EAAEpB,KAAK,CAACoB,GAHE;AAIbC,EAAAA,MAAM,EAAErB,KAAK,CAACqB;AAJD,CAAf","sourcesContent":["import axios from \"axios\";\nimport { toast } from \"react-toastify\";\nconst successCallBack = (success) => {\n  consolle.log(success.data.message);\n  toast(success.data.message);\n};\n\nconst errorCallBack = (error) => {\n  const expectedError =\n    error.response &&\n    error.response.status >= 404 &&\n    error.response.status < 500;\n  if (!expectedError) {\n    console.log(\"logging the error\", error);\n    toast.error(\"an unexpected error occured\");\n  }\n  return Promise.reject(error);\n};\n\n//intercept errors while communicating with backend server\naxios.interceptors.response.use(successCallBack, errorCallBack);\n\nexport default {\n  get: axios.get,\n  post: axios.post,\n  put: axios.put,\n  delete: axios.delete,\n};\n"]},"metadata":{},"sourceType":"module"}